core/templates/dev/head/services/context.service.spec.ts

```diff
@@ -17,362 +17,207 @@
-          }
-        });
-      });
+  describe('behavior in the exploration editor view', () => {
+    beforeEach(() => {
+      ecs = TestBed.get(ContextService);
+      urlService = TestBed.get(UrlService);
+      spyOn(urlService, 'getPathname').and.returnValue('/create/123');
+      spyOn(urlService, 'getHash').and.returnValue('#/gui');
     });
 
-    beforeEach(angular.mock.inject(function($injector) {
-      ecs = $injector.get('ContextService');
-    }));
-
-    it('should correctly retrieve the exploration id', function() {
+    it('should correctly retrieve the exploration id', () => {
       expect(ecs.getExplorationId()).toBe('123');
     });
 
-    it('should correctly retrieve the page context', function() {
+    it('should correctly retrieve the page context', () => {
       expect(ecs.getPageContext()).toBe('editor');
     });
 
-    it('should correctly retrieve exploration editor mode', function() {
+    it('should correctly retrieve exploration editor mode', () => {
       expect(ecs.isInExplorationEditorMode()).toBe(true);
     });
 
-    it('should correctly retrieve the correct entity type', function() {
+    it('should correctly retrieve the correct entity type', () => {
       expect(ecs.getEntityType()).toBe('exploration');
     });
   });
 
-  describe('behavior in the topic editor view', function() {
-    var ecs = null;
-
-    beforeEach(function() {
-      angular.mock.module(function($provide) {
-        $provide.value('UrlService', {
-          getPathname: function() {
-            return '/topic_editor/123';
-          },
-          getHash: function() {
-            return '';
-          }
-        });
-      });
+  describe('behavior in the topic editor view', () => {
+    beforeEach(() => {
+      ecs = TestBed.get(ContextService);
+      urlService = TestBed.get(UrlService);
+      spyOn(urlService, 'getPathname').and.returnValue('/topic_editor/123');
+      spyOn(urlService, 'getHash').and.returnValue('');
     });
 
-    beforeEach(angular.mock.inject(function($injector) {
-      ecs = $injector.get('ContextService');
-    }));
-
-    it('should correctly set editor context to topic editor', function() {
+    it('should correctly set editor context to topic editor', () => {
       ecs.init('topic_editor');
       expect(ecs.getEditorContext()).toBe('topic_editor');
     });
 
-    it('should correctly retrieve the topic id', function() {
+    it('should correctly retrieve the topic id', () => {
       expect(ecs.getEntityId()).toBe('123');
     });
 
-    it('should correctly retrieve the entity type', function() {
+    it('should correctly retrieve the entity type', () => {
       expect(ecs.getEntityType()).toBe('topic');
     });
 
-    it('should correctly retrieve the page context', function() {
+    it('should correctly retrieve the page context', () => {
       expect(ecs.getPageContext()).toBe('topic_editor');
     });
   });
 
-  describe('behavior in question editor modal', function() {
```

Comment:
I don't see this test being replicated. If it wasn't, can you add these tests as well?

Reply:
Ok Done!